packages/@aws-cdk/aws-apigatewayv2/lib/http/route.ts

```diff
@@ -183,11 +204,53 @@ export class HttpRoute extends Resource implements IHttpRoute {
       routeKey: props.routeKey.key,
       target: `integrations/${integration.integrationId}`,
       authorizerId: authBindResult?.authorizerId,
-      authorizationType: authBindResult?.authorizationType ?? 'NONE', // must be explicitly NONE (not undefined) for stack updates to work correctly
+      authorizationType: Lazy.string({
+        produce: () => {
+          if (this.iamEnabled) {
+            return 'AWS_IAM';
+          } else {
+            return authBindResult?.authorizationType ?? 'NONE'; // must be explicitly NONE (not undefined) for stack updates to work correctly
+          }
+        },
+      }),
       authorizationScopes,
     };
 
     const route = new CfnRoute(this, 'Resource', routeProps);
     this.routeId = route.ref;
   }
+
+  /**
+   * Enable IAM authorization on the route.
+   */
+  public enableIamAuthorization(): void {
+    if (this.authorizer) {
+      throw new Error('authorizationType has been set, so we cannot enable AWS_IAM authorization');
+    }
+
+    this.iamEnabled = true;
+  }
+
+  public grantInvoke(grantee: iam.IGrantable, options: GrantInvokeOptions = {}): iam.Grant {
+    this.enableIamAuthorization();
+
+    const httpMethod = options.httpMethod ?? HttpMethod.ANY;
+    const stage = '*';
+    const method = httpMethod === HttpMethod.ANY ? '*' : httpMethod;
+
+    const path = this.path ?? '/';
+
+    // When the user has provided a path with path variables, we replace the
+    // path variable and the rest of the path with a wildcard.
+    const pathSansVariables = path.replace(/\{.*?\}.*/, '');
```

Comment:
This feels slightly magical, I would probably bring this out into a small fn and add tests.

Reply:
I've simplified the expression slightly in the latest commits but haven't pulled it out into a small function. However, I've added a test at the construct level, which ensures that this works correctly.

I could pull this expression out into a small function, but I'd need to export it to test it and annotate it as internal. When working on another area of the CDK, the reviewer advised against this. But, my opinions here aren't strong at all.